### PR TITLE
Fix incorrect comments about atten mask for pytorch backend

### PR DESCRIPTION
**Pull request by lygztq**
# What does this PR do?

I found these comments do not match the actual behavior. In these comments we use `-10000.0` to mask out elements before `softmax` but the actual value used is `torch.finfo(dtype).min`.

## Before submitting
- [x] This PR fixes a typo or improves the docs (you can dismiss the other checks if that's the case).
- [x] Did you read the [contributor guideline](https://github.com/huggingface/transformers/blob/main/CONTRIBUTING.md#start-contributing-pull-requests),
      Pull Request section?
- [ ] Was this discussed/approved via a Github issue or the [forum](https://discuss.huggingface.co/)? Please add a link
      to it if that's the case.
- [ ] Did you make sure to update the documentation with your changes? Here are the
      [documentation guidelines](https://github.com/huggingface/transformers/tree/main/docs), and
      [here are tips on formatting docstrings](https://github.com/huggingface/transformers/tree/main/docs#writing-source-documentation).
- [ ] Did you write any new necessary tests?

## Who can review?

Anyone in the community is free to review the PR once the tests have passed. Feel free to tag
members/contributors who may be interested in your PR.

### CODE REVIEW

**Comment by ydshieh** [comment]
@lygztq Great catch. This inconsistency was introduced in (my) PR #17306, where the docstrings were not updated - my bad.

**Comment by ydshieh** [comment]
I think it's more clear to change from `... the smallest value ...` to `... the dtype's smallest value ...`. WDYT?

**Comment by lygztq** [comment]
> I think it's more clear to change from `... the smallest value ...` to `... the dtype's smallest value ...`. WDYT?

I agree, this is better.

**Review by ydshieh — APPROVED** [review]
Hi @lygztq Sorry for being so late for this PR. I updated the PR with the `main` branch, and with an update for `CodeGen` model. Once the tests pass, we are ready to merge the PR. Thank you for the fix 🤗 !

**Review by sgugger — APPROVED** [review]
Thanks a lot for fixing!
